src/repr/src/scalar.rs

```diff
@@ -3623,52 +3623,44 @@ impl Arbitrary for ScalarType {
-                    // Record
-                    {
-                        // Now we have to use `inner` to create a Record type. First we
-                        // create strategy that creates ColumnType.
-                        let column_type_strat =
-                            (inner, any::<bool>()).prop_map(|(scalar_type, nullable)| ColumnType {
-                                scalar_type,
-                                nullable,
-                            });
-
-                        // Then we use that to create the fields of the record case.
-                        // fields has type vec<(ColumnName,ColumnType)>
-                        let fields_strat =
-                            prop::collection::vec((any::<ColumnName>(), column_type_strat), 0..10);
-
-                        // Now we combine it with the default strategies to get Records.
-                        (fields_strat, any::<Option<GlobalId>>())
-                            .prop_map(|(fields, custom_id)| ScalarType::Record {
-                                fields,
-                                custom_id,
-                            })
-                            .boxed()
-                    },
-                ])
-            },
-        )
+        leaf.prop_recursive(2, 3, 5, |inner| {
+            Union::new(vec![
+                // List
+                (inner.clone(), any::<Option<GlobalId>>())
+                    .prop_map(|(x, id)| ScalarType::List {
+                        element_type: Box::new(x),
+                        custom_id: id,
+                    })
+                    .boxed(),
+                // Map
+                (inner.clone(), any::<Option<GlobalId>>())
+                    .prop_map(|(x, id)| ScalarType::Map {
+                        value_type: Box::new(x),
+                        custom_id: id,
+                    })
+                    .boxed(),
+                // Record
+                {
+                    // Now we have to use `inner` to create a Record type. First we
+                    // create strategy that creates ColumnType.
+                    let column_type_strat =
+                        (inner, any::<bool>()).prop_map(|(scalar_type, nullable)| ColumnType {
+                            scalar_type,
+                            nullable,
+                        });
+
+                    // Then we use that to create the fields of the record case.
+                    // fields has type vec<(ColumnName,ColumnType)>
+                    let fields_strat =
+                        prop::collection::vec((any::<ColumnName>(), column_type_strat), 0..10);
+
+                    // Now we combine it with the default strategies to get Records.
+                    (fields_strat, any::<Option<GlobalId>>())
+                        .prop_map(|(fields, custom_id)| ScalarType::Record { fields, custom_id })
+                        .boxed()
+                },
+            ])
+        })
```

Comment:
nothing changed in this entire block other than changing a `4` to `3` and removing a comment. Removing the (incorrect) comment caused the block to get reformatted